code/workspaces/auth-service/src/controllers/userManagement.spec.js

```diff
@@ -104,58 +102,49 @@ describe('user management controller', () => {
     });
   });
 
-  describe('setInstanceAdmin', () => {
+  describe('setSystemRole', () => {
     it('should return roles as JSON', async () => {
       // Arrange
       const request = httpMocks.createRequest();
+      request.params = { userId: '123' };
+      request.body = { instanceAdmin: true };
       const response = httpMocks.createResponse();
+      setSystemRoleMock.mockResolvedValue(roles);
 
       // Act
-      await userManagement.setInstanceAdmin(request, response);
+      await userManagement.setSystemRole(request, response);
 
       // Assert
+      expect(setSystemRoleMock).toBeCalledWith('123', 'instanceAdmin', true);
       expect(response.statusCode).toBe(200);
       expect(response._getData()) // eslint-disable-line no-underscore-dangle
         .toEqual(roles);
     });
 
-    it('should return 500 if error', async () => {
+    it('should return 400 if invalid input', async () => {
       // Arrange
-      setInstanceAdminMock.mockRejectedValue('no such catalogue');
       const request = httpMocks.createRequest();
+      request.params = { userId: '123' };
+      request.body = {};
       const response = httpMocks.createResponse();
+      setSystemRoleMock.mockResolvedValue(roles);
 
       // Act
-      await userManagement.setInstanceAdmin(request, response);
+      await userManagement.setSystemRole(request, response);
 
       // Assert
-      expect(response.statusCode).toBe(500);
+      expect(response.statusCode).toBe(400);
     });
-  });
-
-  describe('setCatalogueRole', () => {
-    it('should return roles as JSON', async () => {
-      // Arrange
-      const request = httpMocks.createRequest();
-      const response = httpMocks.createResponse();
-
-      // Act
-      await userManagement.setCatalogueRole(request, response);
-
-      // Assert
-      expect(response.statusCode).toBe(200);
-      expect(response._getData()) // eslint-disable-line no-underscore-dangle
-        .toEqual(roles);
-    });
-
     it('should return 500 if error', async () => {
       // Arrange
-      setCatalogueRoleMock.mockRejectedValue('no such catalogue');
       const request = httpMocks.createRequest();
+      request.params = { userId: '123' };
+      request.body = { instanceAdmin: true };
       const response = httpMocks.createResponse();
+      setSystemRoleMock.mockRejectedValue('no such catalogue');
```

Comment:
Does this rejected value still make sense now it is not a function for setting the catalogue permission?